stdlib/test/builtin/test_list.mojo

```diff
@@ -39,7 +39,28 @@ fn test_repr_list() raises:
     assert_equal(empty.__repr__(), "[]")
 
 
+fn test_equality() raises:
+    var p = List(2, 3, 5, 7)
```

Comment:
**Suggestion** I recommend also testing the `List.__contains__` with memory-only types, such as `String`.  Of note, this is homogenous whereas, the `ListLiteral` test case is a collection of heterogenous elements (as expected).

Reply:
On this note, shall I move the `List.__contains__` testing to [`collections/test_list.mojo`](https://github.com/modularml/mojo/blob/nightly/stdlib/test/builtin/test_list.mojo) rather?

Reply:
Well, the `List` already has tests for `contains` - see https://github.com/modularml/mojo/blob/a78d7548ebebfc074d237c21da29e027c89fd3f1/stdlib/test/collections/test_list.mojo#L792.  I'd recommend we beefen up those tests to include `List[String]` so we test a memory-only type.

I'd keep the `ListLiteral.__contains__` tests here in `stdlib/test/builtin/test_list.mojo`.  How does that sound?

Reply:
Happy :)